packages/react-reconciler/src/ReactFiberLane.js

```diff
@@ -705,6 +738,21 @@ export function markRootFinished(root: FiberRoot, remainingLanes: Lanes) {
   }
 }
 
+export function markRootEntangled(root: FiberRoot, entangledLanes: Lanes) {
+  root.entangledLanes |= entangledLanes;
+
+  const entanglements = root.entanglements;
+  let lanes = entangledLanes;
+  while (lanes > 0) {
+    const index = ctrz(lanes);
+    const lane = 1 << index;
+
+    entanglements[index] |= lanes;
+
+    lanes &= ~lane;
```

Comment:
This mechanism of clearing each lane after we've processed it- so the next call to `ctrz ` advances the index- does it actually cause a bug in this case?

The result of this loop is that each processed lane isn't marked as being entangled with the preceding lanes, only the following ones.

I *think* that means we could miss updates in the check above?

Reply:
Yeah I just noticed this while working on a PR on top. Good catch!